python/docs/source/guide/execution.md

```diff
@@ -0,0 +1,90 @@
+# Execution
+
+A Timestream may be executed and written to a [destination](#destinations) in a variety of ways.
+The things to consider when deciding how to execute the Timestream are:
+
+1. Whether you want the _history_ of points or the _snapshot_ of values for each entity at a given time.
+2. Whether you want to run a single _query_ or start a _materialization_.
+3. Whether you want to limit the output to points/rows that have changed since a given time.
+4. Whether you want to stop at a given point in time.
+
+[State](#state) can be used to provide fault-tolerance and allow incremental processing of only new events.
+
+```{admonition} Preview during development
+:class: tip
+
+While developing queries, it is often useful to view a few rows from the result.
+Using {py:meth}`kaskada.Timestream.preview` you can retrieve a small set of rows from the result set as a Pandas DataFrame.
+```
+
+## History vs. Snapshot
+
+Executing a Timestream for the history outputs every point in the Timestream.
```

Comment:
Link to `Timestream` explanation, if we have one? Maybe not here, but somewhere in the page.